src/libs/actions/ReimbursementAccount/fetchFreePlanVerifiedBankAccount.js

```diff
@@ -0,0 +1,219 @@
+            const bankAccountJSON = _.find(bankAccountList, account => (
+                account.bankAccountID === bankAccountID
+            ));
+            const bankAccount = bankAccountJSON ? new BankAccount(bankAccountJSON) : null;
+            const throttledHistoryCount = lodashGet(nameValuePairs, CONST.NVP.BANK_ACCOUNT_GET_THROTTLED, 0);
+            const isPlaidDisabled = throttledHistoryCount > CONST.BANK_ACCOUNT.PLAID.ALLOWED_THROTTLED_COUNT;
+
+            return {
+                maxAttemptsReached,
+                kycVerificationsMigration,
+                throttledDate,
+                bankAccount,
+                isPlaidDisabled,
+                bankAccountID,
+            };
+        })
+        .finally(() => {
+            Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: false});
+        });
+}
+
+/**
+ * @param {BankAccount} bankAccount
+ * @param {String} kycVerificationsMigration
+ * @returns {Boolean}
+ */
+function getHasTriedToUpgrade(bankAccount, kycVerificationsMigration) {
+    if (!bankAccount) {
+        return false;
+    }
+
+    return bankAccount.getDateSigned() > (kycVerificationsMigration || '2020-01-13');
+}
+
+/**
+ * @param {String} stepToOpen
+ * @param {String} stepFromStorage
+ * @param {Object} achData
+ * @param {BankAccount} bankAccount
+ * @param {Boolean} hasTriedToUpgrade
+ * @returns {String}
+ */
+function getCurrentStep(stepToOpen, stepFromStorage, achData, bankAccount, hasTriedToUpgrade) {
+    // If we are providing a stepToOpen via a deep link then we will always navigate to that step. This
+    // should be used with caution as it is possible to drop a user into a flow they can't complete e.g.
+    // if we drop the user into the CompanyStep, but they have no accountNumber or routing Number in
+    // their achData.
+    if (stepToOpen) {
+        return stepToOpen;
+    }
+
+    // To determine if there's any step we can go to we will look at the data from the server first then whatever is in device storage.
+    const currentStep = achData.currentStep
+        ? getNextStepToComplete(achData)
+        : getReimbursementAccountInSetup().currentStep;
+
+    if (achData.isInSetup) {
+        return currentStep || CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+    }
+
+    // If we don't have a bank account then take the user to the BankAccountStep so they can create one.
+    if (!bankAccount) {
+        return CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+    }
+
+    if (bankAccount.isPending() || bankAccount.isVerifying()) {
+        return CONST.BANK_ACCOUNT.STEP.VALIDATION;
+    }
+
+    // No clear place to direct this user so we'll go with the bank account step
+    if (!bankAccount.isOpen()) {
+        return CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+    }
+
+    if (bankAccount.needsToPassLatestChecks()) {
+        return hasTriedToUpgrade ? CONST.BANK_ACCOUNT.STEP.VALIDATION : CONST.BANK_ACCOUNT.STEP.COMPANY;
+    }
+
+    return CONST.BANK_ACCOUNT.STEP.ENABLE;
```

Comment:
I'm unsure about defaulting to the `BankAccountStep` because if we miss a case where the user should be passed to the enable step then they'll get stuck and we won't might not realize that we missed this case. 

Maybe we can leave the logic intact and log an alert when we end up navigating to this step for an unexpected reason?

Reply:
Something like

```js
if (!shouldGoToEnableStep) {
    Log.alert('Unexpectedly sent user to EnableStep', params);
}

return CONST.BANK_ACCOUNT.STEP.ENABLE;
```

Reply:
Sounds good!

Reply:
Ah actually, thought about it some more and I think the only possibility is that the bank account is not in an open state, but we are already returning early above when we don't have a bank account and it is not in an open state. So I think it might just be redundant to do anything else. 

Can you think of a case where we end up here and it would not be safe to go to the enable step?

Maybe it would be best to wait for that day to arrive...

Reply:
I can't think of any. Given that this would be a rare edge case, if it exists at all, I'm ok keeping it as is.